### PR TITLE
Changed older versions button to links

### LINKED ISSUE #9811: Show download links instead of install buttons in all versions page
### Describe the problem and steps to reproduce it:

1. Install NoScript (possibly an older version).
2. Visit https://addons.mozilla.org/en-US/firefox/addon/noscript/versions/ and/or
  https://addons.mozilla.org/en-US/firefox/addon/noscript/
3. Try to install an older version, or at least find a "Download file" link (from #4910).

### What happened?
I am not able to install or download an older version. All buttons have changed to "Remove".

I cannot upgrade to the latest version via the addon listing because there is only a "Remove" button.

### What did you expect to happen?
It should be possible to install or download an older version.

### Anything else we should know?

Actually, it is possible to download, by right-clicking on the button and opening it in a new tab/window, because the button is actually a link. This is however not obvious and not a great user experience.

### PR DESCRIPTION
**Pull request by smitbarmase**
Fixes #9811

This contains fixes from @sarthakkundra from #9973 and tests from @smitbarmase from #10098 

Recent changes contain:
1. Added test to verify that the `AMInstallButton` is not displayed, in older versions on Firefox browser.
2. Omit the property `showLinkInsteadOfButton` when it is false.
3. Versions page using a non-Firefox browser, no longer show "download Firefox" buttons for each version.

I refactored the code a bit to fix it and make it look cleaner and more understandable.

I have fixed all the bugs suggested in #10181 

@bobsilverberg PR is ready for review!

### CODE REVIEW

File: tests/unit/amo/components/TestInstallButtonWrapper.js [hunk 1]
```diff
@@ -328,23 +328,63 @@ describe(__filename, () => {
...
 
-  it('does not display a download link when the browser is compatible', () => {
+  it('does not display a download link when the browser is compatible and showLinkInsteadOfButton is false', () => {
     const _findInstallURL = sinon
       .stub()
       .returns('https://a.m.o/files/addon.xpi');
     const _getClientCompatibility = sinon.stub().returns({
       compatible: true,
     });
+    const showLinkInsteadOfButton = false;
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestInstallButtonWrapper.js:338]: You don't really need a `const` for this. You could just include `showLinkInsteadOfButton: false` on line 344 below.

File: tests/unit/amo/components/TestInstallButtonWrapper.js [hunk 2]
```diff
@@ -328,23 +328,63 @@ describe(__filename, () => {
...
   });
 
+  it('displays a download link when the browser is compatible and showLinkInsteadOfButton is true', () => {
+    const _findInstallURL = sinon
+      .stub()
+      .returns('https://a.m.o/files/addon.xpi');
+    const _getClientCompatibility = sinon.stub().returns({
+      compatible: true,
+    });
+    const showLinkInsteadOfButton = true;
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestInstallButtonWrapper.js:357]: As above, we don't really need this `const`.

File: tests/unit/amo/components/TestInstallButtonWrapper.js [hunk 3]
```diff
@@ -328,23 +328,63 @@ describe(__filename, () => {
...
+  });
+
+  it('does not display a button when the browser is compatible and showLinkInsteadOfButton is true', () => {
+    const _findInstallURL = sinon
+      .stub()
+      .returns('https://a.m.o/files/addon.xpi');
+    const _getClientCompatibility = sinon.stub().returns({
+      compatible: true,
+    });
+    const showLinkInsteadOfButton = true;
```
Comment (bobsilverberg) [inline tests/unit/amo/components/TestInstallButtonWrapper.js:376]: As above, no need for this `const`.

**Review by bobsilverberg — CHANGES_REQUESTED** [review]
Looks good, nice updates @smitbarmase! I just added a few comments about the tests.

**Comment by smitbarmase** [comment]
@bobsilverberg I have made the changes.

**Review by bobsilverberg — CHANGES_REQUESTED** [review]
Thanks for the updates @smitbarmase, they look good. I did notice one problem on the all versions page, though. The "Download file" link is left justified with the button for the latest version, but right justified in the column for the rest, which looks a bit ugly. Can you see if you can fix that? Please be sure to not change the way it looks on the add-on detail page, though.

Here's an example:

Screen Shot 2021-03-26 at 8 25 59 AM

**Comment by bobsilverberg** [comment]
@smitbarmase Just checking in to see if you think you can address the comment added above. Thanks!

**Comment by smitbarmase** [comment]
@bobsilverberg Yes, I can. I was quite busy this week as I have exams. Sorry for that.

One thing more, while running the project on the local server, it does not load SCSS files. So any styling does not appear on the window. All I see is HTML. I don't know why this is happening. Any suggestions?

I'm using WSL 2 on Windows 10.

**Comment by bobsilverberg** [comment]
> @bobsilverberg Yes, I can. I was quite busy this week as I have exams. Sorry for that.
> 
> One thing more, while running the project on the local server, it does not load SCSS files. So any styling does not appear on the window. All I see is HTML. I don't know why this is happening. Any suggestions?
> 
> I'm using WSL 2 on Windows 10.

I'm not sure what might be causing that. Can you post any messages you see in the browser console that might indicate what the problem is?

**Comment by bobsilverberg** [comment]
@smitbarmase We have another change landing next week that will really benefit from having this in place. Are you able to address the final concern raised? If not, then I can make the final changes and land this.

**Comment by smitbarmase** [comment]
Here is what I'm getting. No CSS files have loaded. And there is one error in the console.

Browser console:
image

Screenshot:
image

I can't see any type of error in node console tough.

Node console:
```
[0] orphan modules 43.8 KiB [orphan] 29 modules
[0] runtime modules 31.1 KiB 16 modules
[0] modules by path ./src/ 7.08 MiB
[0]   modules by path ./src/amo/ 3.06 MiB 513 modules
[0]   modules by path ./src/locale/ 4.01 MiB 126 modules
[0]   modules by path ./src/fonts/ 12.5 KiB
[0]     modules by path ./src/fonts/woff2/*.woff2 840 bytes 10 modules
[0]     modules by path ./src/fonts/otf/*.otf 656 bytes 8 modules
[0]     modules by path ./src/fonts/*.scss 11 KiB
[0]       ./src/fonts/inter.scss 1.61 KiB [built] [code generated]
[0]       ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[1].use[1]!./node_modules/postcss-loader/dist/cjs.js??ruleSet[1].rules[1].use[2]!./node_modules/sass-loader/dist/cjs.js!./src/fonts/inter.scss 9.38 KiB [built] [code generated]
[0] modules by path ./node_modules/ 3.31 MiB 493 modules
[0] ./src/locale/ sync ./node_modules/bundle-loader/index.js?name=[name]-i18n-[folder]! ^\.\/.*?\/amo\.js$ 1.18 KiB [built] [code generated]
[0] webpack 5.17.0 compiled successfully in 55273 ms
[1] [1617945375746404400] INFO  (amo/947 on comet): 🔥 Addons-frontend server is running [ENV:development] [isDevelopment:true] [isDeployed:false] [apiHost:http://localhost:3000] [apiPath:/api/] [apiVersion:v5]
[1] [1617945375748492300] DEBUG (amo/947 on comet): 🚦 Proxy detected, frontend running at http://127.0.0.1:3333.
[1] [1617945375748664600] DEBUG (amo/947 on comet): 👁  Open your browser at http://127.0.0.1:3000 to view it.
[1] 👀 The file watcher for server hot reload is ready.
```

**Comment by bobsilverberg** [comment]
@smitbarmase What URL are you visiting when trying to load the site locally in your browser?

**Comment by bobsilverberg** [comment]
@smitbarmase It would be nice to land this change for next week, so just wondering if you are able to address the last comment on it. If you don't think you'll have time I can finish it off for you.

**Comment by smitbarmase** [comment]
@bobsilverberg Sure. Thank you for that :)

**Comment by bobsilverberg** [comment]
This was replaced by #10401

**Pull request closed without merging**